pytket/phir/phirgen.py

```diff
@@ -218,13 +219,150 @@ def convert_gate(op: tk.Op, cmd: tk.Command) -> JsonDict | None:
+                    cond = {
+                        "cop": ">=",
+                        "args": [args[0].reg_name, l],
+                    }
+                case 0, u:
+                    cond = {
+                        "cop": "<=",
+                        "args": [args[0].reg_name, u],
+                    }
+            out = {
+                "block": "if",
+                "condition": cond,
+                "true_branch": [assign_cop([arg_to_bit(cmd.bits[0])], [1])],
+            }
+        case tk.MultiBitOp():
+            cop = cop_from_op_name(op.basic_op.get_name())
+            is_explicit = op.basic_op.type == tk.OpType.ExplicitPredicate
+
+            # determine number of register operands involved in the operation
+            operand_count = len(args) // len(cmd.bits) - (1 if is_explicit else 0)
+
+            iters = [iter(args)] * (operand_count + (1 if is_explicit else 0))
+            iter2 = deepcopy(iters)
+
+            # Columns of expressions, e.g.,
+            #   AND (*2) a[0], b[0], c[0]
+            #          , a[1], b[1], c[1]
+            #   would be [(a[0], a[1]), (b[0], b[1]), (c[0], c[1])]
+            # and AND (*2) a[0], a[1], b[0]
+            #            , b[1], c[0], c[1]
+            #   would be [(a[0], b[1]), (a[1], c[0]), (b[0], c[1])]
+            cols = zip(*zip(*iters, strict=True), strict=True)
+
+            if all(
+                all(col[0].reg_name == bit.reg_name for bit in col) for col in cols
+            ):  # expression can be applied register-wise
+                out = assign_cop(
+                    [cmd.bits[0].reg_name],
+                    [
+                        {
+                            "cop": cop,
+                            "args": [arg.reg_name for arg in args[:operand_count]],
+                        }
+                    ],
+                )
+            else:  # apply a sequence of bit-wise ops
+                exps = zip(*iter2, strict=True)
+                out = {
+                    "block": "sequence",
+                    "ops": [
+                        assign_cop(
+                            [arg_to_bit(bit)],
+                            [
+                                {
+                                    "cop": cop,
+                                    "args": [
+                                        arg_to_bit(arg) for arg in exp[:operand_count]
+                                    ],
+                                }
+                            ],
+                        )
+                        for bit, exp in zip(cmd.bits, exps, strict=True)
+                    ],
+                }
+        case _:
+            raise NotImplementedError(op)
+
+    return out
+
+
+def multi_bit_condition(args: "list[UnitID]", value: int) -> JsonDict:
+    """Construct bitwise condition."""
+    return {
+        "cop": "&",
+        "args": [
+            {"cop": "==", "args": [arg_to_bit(arg), bval]}
```

Comment:
Need to be careful of endianness here. The conditional `value` in pytket is little-endian, e.g. a value of 2 on the bits `[a[0], a[1]]` means `a[0] == 0 && a[1] == 1`. Does PHIR use the same convention?

Reply:
@qciaran would you be able to confirm. I forgot to mention you at https://github.com/CQCL/pytket-phir/pull/162#discussion_r1555883698 since I also had my doubts.

Reply:
Yes that is the intention: the order in which the bits are given corresponds to the little-endian binary expansion of the value. In your example, the condition is `b[1] == 0 && b[2] == 1`.

Reply:
Yeah, in PHIR `a[0]==0 && a[1]=1` is the integer value 2.

Reply:
Filed https://github.com/CQCL/phir/issues/77, this PR seems ready to be merged.